src/test/java/uk/gov/companieshouse/api/testdata/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,203 @@
+import uk.gov.companieshouse.api.testdata.model.rest.RolesSpec;
+import uk.gov.companieshouse.api.testdata.model.rest.UserTestData;
+import uk.gov.companieshouse.api.testdata.model.rest.UsersSpec;
+import uk.gov.companieshouse.api.testdata.repository.RoleRepository;
+import uk.gov.companieshouse.api.testdata.repository.UserRepository;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceImplTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private RoleRepository roleRepository;
+
+    @InjectMocks
+    private UserServiceImpl userServiceImpl;
+
+    @Test
+    void testCreateUserWithoutRoles() throws DataException {
+        UsersSpec usersSpec = new UsersSpec();
+        usersSpec.setPassword("password");
+
+        Users mockUser = new Users();
+        mockUser.setId("generated-user-id");
+
+        when(userRepository.save(any(Users.class))).thenReturn(mockUser);
+
+        UserTestData userTestData = userServiceImpl.createUser(usersSpec);
+
+        assertNotNull(userTestData.getUserId(), "User ID should not be null");
+        assertNotNull(userTestData.getEmail(), "Email should not be null");
+        assertTrue(userTestData.getForename().contains("Forename"), "Forename should contain Forename");
+        assertTrue(userTestData.getSurname().contains("Surname"), "Surname should contain Surname");
+    }
+
+    @Test
+    void testCreateUserWithRoles() throws DataException {
+        UsersSpec usersSpec = new UsersSpec();
+        usersSpec.setPassword("password");
+
+        RolesSpec roleSpec = new RolesSpec();
+        roleSpec.setId("role-id");
+        roleSpec.setPermissions(Arrays.asList("permission1", "permission2"));
+        usersSpec.setRoles(List.of(roleSpec));
+
+        Roles mockRole = new Roles();
+        mockRole.setId("role-id-playwright-role" + System.currentTimeMillis());
+        mockRole.setPermissions(roleSpec.getPermissions());
+
+        Users mockUser = new Users();
+        mockUser.setId("generated-user-id");
+
+        when(roleRepository.save(any(Roles.class))).thenReturn(mockRole);
+        when(userRepository.save(any(Users.class))).thenReturn(mockUser);
+
+        UserTestData userTestData = userServiceImpl.createUser(usersSpec);
+
+        assertNotNull(userTestData.getUserId(), "User ID should not be null");
+        assertNotNull(userTestData.getEmail(), "Email should not be null");
+        assertTrue(userTestData.getForename().contains("Forename"), "Forename should contain Forename");
+        assertTrue(userTestData.getSurname().contains("Surname"), "Surname should contain Surname");
+
+        // Verify that the role repository was called
+        verify(roleRepository, times(1)).save(any(Roles.class));
```

Comment:
I would assert the values of the `Roles` parameter are right by using an argument captor

Reply:
added argument captor for users and roles for assertion as suggested